pkg/connector/sso_group.go

```diff
@@ -339,27 +351,71 @@ func (g *ssoGroupResourceType) Grant(
 }
 func (g *ssoGroupResourceType) Revoke(ctx context.Context, grant *v2.Grant) (annotations.Annotations, error) {
 	if grant.Principal.Id.ResourceType != resourceTypeSSOUser.Id {
-		return nil, errors.New("baton-aws: only sso users can be removed from sso group")
+		return nil, errors.New("baton-aws: only sso users can be removed from sso groups")
 	}
 
 	l := ctxzap.Extract(ctx).With(
 		zap.String("grant_id", grant.Id),
 		zap.String("identity_store_id", awsSdk.ToString(g.identityInstance.IdentityStoreId)),
 	)
 
+	annos := annotations.New()
+	membershipId := grant.Id
+
 	resp, err := g.identityStoreClient.DeleteGroupMembership(
 		ctx,
 		&awsIdentityStore.DeleteGroupMembershipInput{
 			IdentityStoreId: g.identityInstance.IdentityStoreId,
-			MembershipId:    awsSdk.String(grant.Id),
+			MembershipId:    awsSdk.String(membershipId),
+		},
+	)
+	if err == nil {
+		l.Debug("revoked grant", zap.String("membership_id", membershipId))
+		if reqId := extractRequestID(&resp.ResultMetadata); reqId != nil {
+			annos.Append(reqId)
+		}
+
+		return annos, nil
+	}
+
+	l.Error("aws-connector: Failed to delete group membership. Trying to fetch group membership in case grant ID is incorrect", zap.Error(err))
```

Comment:
I think that since we expect this to happen a lot, this is maybe a log level INFO?